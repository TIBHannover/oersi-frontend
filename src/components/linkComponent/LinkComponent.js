import React from "react"
import "./linkComponent.css"
import PropTypes from "prop-types"

/**
 * @author Edmond Kacaj <edmondikacaj@gmail.com>
 * @param {*} props properties
 */
const LinkComponent = (props) => {
  return (
    <div>
      <a
        className="link-component"
        target="_blank"
        rel="noopener noreferrer"
        href={props.link}
      >
        {props.children}
      </a>
    </div>
  )
}

LinkComponent.propTypes = {
  data: PropTypes.exact({
    link: PropTypes.string,
    text: PropTypes.string,
  }),
}

export default LinkComponent
