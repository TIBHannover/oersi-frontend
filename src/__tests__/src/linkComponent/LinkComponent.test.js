import React from "react"
import ReactDOM from "react-dom"
import LinkComponent from "../../../components/linkComponent/LinkComponent"

describe("LinkComponent ==> Test UI  ", () => {
  it("LinkComponent : should render without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(
      <LinkComponent link={"http://test.com"}>
        <span>Test</span>
      </LinkComponent>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})
