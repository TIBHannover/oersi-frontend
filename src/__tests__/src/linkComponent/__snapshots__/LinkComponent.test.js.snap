// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkComponent ==> Test UI   should render correctly 1`] = `ShallowWrapper {}`;
